### ISSUE TITLE
kubernetes_secret and kubernetes_storage_class unable to refresh state

### ISSUE THREAD

**Issue by apeabody** [terraform-google-modules/terraform-google-gke-gitlab#97]
### TL;DR

The `kubernetes_secret` and `kubernetes_storage_class` are unable to refresh state.  This manifests as the kubernetes provider attempting to dial tcp 127.0.0.1:80 rather than the [configured](https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/1971baaf005070bf971e2b401293ec15f11b466f/main.tf#L56) cluster.  This may be a result of the order in which Terraform evaluates the provider block vs the resources as noted in the [Kubernetes Provider documentation](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#stacking-with-managed-kubernetes-cluster-resources) and [GKE example](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/_examples/gke/README.md).

This also blocks #90 as with TF 0.15+ a refresh state is attempted prior to the destroy phase.

### Terraform Configuration

```hcl
CI
```

### Terraform Version

```sh
TF 0.15+
```
